### PR TITLE
[#962][#980] AKS base cluster deploy with Terraform

### LINKED ISSUE #962: Terraform module for AKS setup
As A DevOps engineer I'd like to have terraform modules for AKS cluster setup.

### LINKED ISSUE #980: Terraform module for K8S setup at AKS
As a DevOps engineer I'd like to have terraform modules for Legion K8S dependencies setup at AKS cluster so I can deploy Legion components

### PR DESCRIPTION
**Pull request by maxkochubey**
Set of terraform modules for basic cluster installation.

Features:
- Before the deployment, it is necessary to provide a number of preconditions described in the corresponding [README](https://github.com/legion-platform/legion-infrastructure/blob/feat/962-terraform-aks/terraform/env_types/azure/aks/README.md).
  - Public IP for AKS cluster (to provide static ingress entrypoint) and initial resources group must be created in advance.
  - Blob storage for terraform states should be created manually in the same resource group.

Closes [#962](https://github.com/legion-platform/legion/issues/962) and closes [#980](https://github.com/legion-platform/legion/issues/980).

### CODE REVIEW

File: terraform/env_types/azure/aks/aks_create/variables.tf [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    #   }
+    #   node_config = {
+    #     machine_type     = "Standard_B8ms"
+    #     disk_size_gb     = "100"
+    #     labels = {
+    #       mode           = "legion-training"
+    #     }
+    #     taint = [{
+    #       key    = "dedicated"
+    #       value  = "training"
```
Comment (karbyshevds) [inline terraform/env_types/azure/aks/aks_create/variables.tf]: please, remove commented code

**Review by mcdoker18 — APPROVED** [review]

File: terraform/env_types/azure/aks/k8s_setup/main.tf [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+# TBD:
+# Make appropriate analogue of GKE Service Account Assigner mechanism for AKS
+#
+# module "gke-saa" {
+#   source                    = "../../../../modules/k8s/gke-saa"
+#   legion_helm_repo          = var.legion_helm_repo
+#   legion_infra_version      = var.legion_infra_version
+# }
```
Comment (karbyshevds) [inline terraform/env_types/azure/aks/k8s_setup/main.tf]: please remove gke-saa declaration

**Review by mcdoker18 — APPROVED** [review]

File: terraform/env_types/azure/aks/legion/main.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#   zone                     = var.zone
+#   region                   = var.region
+#   project_id               = var.project_id
+#   legion_helm_repo         = var.legion_helm_repo
+#   root_domain              = var.root_domain
+#   docker_repo              = var.docker_repo
+#   docker_user              = var.docker_user
+#   docker_password          = var.docker_password
+#   legion_version           = var.legion_version
+#   collector_region         = var.collector_region
```
Comment (karbyshevds) [inline terraform/env_types/azure/aks/legion/main.tf]: please, remove commented code

File: terraform/modules/azure/aks_cluster/main.tf [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  api_server_authorized_ip_ranges = []
+
+  network_profile {
+    network_plugin = "azure"
+    network_policy = "calico"
+    #pod_cidr: ""
+    #service_cidr: "10.0.0.0/16"
+    #dns_service_ip: "10.0.0.10"
+    #docker_bridge_cidr: "172.17.0.1/16"
+    #load_balancer_sku: "basic"
```
Comment (karbyshevds) [inline terraform/modules/azure/aks_cluster/main.tf]: please, remove commented code

**Review by karbyshevds — CHANGES_REQUESTED** [review]

File: terraform/env_types/azure/aks/helm_init/variables.tf [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+############################################################################################################
+
+variable "cluster_name" {
+  description = "Legion k8s cluster name"
+  default     = "legion"
+}
+
+variable "tiller_image" {
+  default = "gcr.io/kubernetes-helm/tiller:v2.14.0"
```
Comment (mcdoker18) [inline terraform/env_types/azure/aks/helm_init/variables.tf]: Upgrade version

**Review by mcdoker18 — CHANGES_REQUESTED** [review]

**Review by mcdoker18 — APPROVED** [review]

File: terraform/env_types/azure/aks/aks_create/variables.tf (hunk 1 above)
Comment (maxkochubey) [reply terraform/env_types/azure/aks/aks_create/variables.tf]: Fixed in bd2f130.

File: terraform/env_types/azure/aks/k8s_setup/main.tf (hunk 2 above)
Comment (maxkochubey) [reply terraform/env_types/azure/aks/k8s_setup/main.tf]: Fixed in bd2f130.

File: terraform/env_types/azure/aks/legion/main.tf (hunk 3 above)
Comment (maxkochubey) [reply terraform/env_types/azure/aks/legion/main.tf]: Fixed in bd2f130.

File: terraform/modules/azure/aks_cluster/main.tf (hunk 4 above)
Comment (maxkochubey) [reply terraform/modules/azure/aks_cluster/main.tf]: Fixed in bd2f130.

**Review by karbyshevds — APPROVED** [review]

**Pull request merged**